File: algorithm/leetcode831.masking-personal-information.cpp
class Solution {
public:
    string maskPII(string S) {
    	int at = S.find("@");
    	string ans;
    	if(at != string::npos)
    	{
    		if(isupper(S[0]))
    			ans = string(1,S[0]+32)+"*****";
    		else
    			ans = string(1,S[0])+"*****";
    		for(int i = at-1; i < S.size(); ++i)
    		{
    			if(isupper(S[i]))
    				S[i] += 32;
    			ans += S[i];
    		}
    	}
        else
        {
            int count = 0;
            string last4;
            for(int i = S.size()-1; i >= 0; --i)
            {
                if(isdigit(S[i]))
                {
                    count++;
                    if(last4.size()<4)
                        last4 = S[i]+last4;
                }
            }
            if(count<=10)
                ans = "***-***-"+last4;
            else
                ans = "+"+string(count-10,'*')+"-***-***-"+last4;
        }
    	return ans;
    }
};

File: algorithm/Python3_version/leetcode831.masking-personal-information.py
class Solution:
    def maskPII(self, S: str) -> str:
        at = '@' in S
        ans = ""
        if at:
            if S[0].isupper():
                ans = S[0].lower()+"*****"
            else:
                ans = S[0]+"*****"
            idx = S.index('@')
            for i in range(idx-1, len(S)):
                if S[i].isupper():
                    ans += S[i].lower()
                else:
                    ans += S[i]
        else:
            count = 0
            last4 = ""
            for i in range(len(S)-1,-1,-1):
                if S[i].isdigit():
                    count += 1
                    if len(last4)<4:
                        last4 = S[i]+last4
            if count==10:
                ans = "***-***-"+last4
            else:
                ans = "+"+'*'*(count-10)+"-***-***-"+last4
        return ans